pgxn/neon/libpagestore.c

```diff
@@ -844,6 +849,16 @@ pg_init_libpagestore(void)
 							PGC_USERSET,
 							0,	/* no flags required */
 							NULL, (GucIntAssignHook) &readahead_buffer_resize, NULL);
+	DefineCustomIntVariable("neon.protocol_version",
+							"Version of compute<->page server protocol",
+							NULL,
+							&neon_protocol_version,
+							1, /* default to old protocol for now */
+							1, /* min */
+							2, /* max */
+							PGC_SU_BACKEND,
```

Comment:
You cannot change it on-the-fly within the backend. Different backends can have different settings, but after the backend has launched, the setting cannot be changed. (That would require reconnecting to the pageserver with the new version.)

PGC_SU_BACKEND seems right for that. It allows a superuser to set the value at backend startup, in the startup packet. And you can change it in postgresql.conf and reload; it will then take effect in new backends, but old backends will continue with the old value.